### PR TITLE
email upload error for bulk users

### PR DESCRIPTION
**Pull request by benrudolph**
screen shot 2016-07-07 at 11 50 06 pm

cc: @proteusvacuum @sravfeyn @esoergel (i forgot my original buddy is back online!)

### CODE REVIEW

**Comment by benrudolph** [comment]
http://manage.dimagi.com/default.asp?231796

File: corehq/apps/users/bulkupload.py [hunk 1]
```diff
@@ -487,12 +487,14 @@ def is_password(password):
...
-                        email_validator = EmailValidator()
                         try:
-                            email_validator(email)
-                            user.email = email
+                            validate_email(email)
                         except ValidationError:
                             raise UserUploadError(_("User has an invalid email address"))
+
+                        if not email.islower():
+                            raise UserUploadError(_("Email address must be lower case: {}".format(email)))
```
Comment (proteusvacuum) [inline corehq/apps/users/bulkupload.py]: I know the ticket asks for a better error message... but what is the reason we don't allow caps in email addresses? 
Can we not just force the email address to be lower case, and not make this an error condition?

Comment (benrudolph) [inline corehq/apps/users/bulkupload.py]: i originally looked into doing that and i think in 99.9% of the time that's fine. there were some people noting that in rare situations caps could make a difference. i think looking at this again, it's probably not worth having people go through the hassle for that 0.001% change

Comment (proteusvacuum) [inline corehq/apps/users/bulkupload.py]: out of curiosity, what were those rare situations?

Comment (benrudolph) [inline corehq/apps/users/bulkupload.py]: http://stackoverflow.com/questions/9807909/are-email-addresses-case-sensitive

take a look at that answer and the comments. not the most in depth research, so maybe i'll dive in a bit more to see if i can get something more definitive

Comment (benrudolph) [inline corehq/apps/users/bulkupload.py]: actually a bunch of the answers are somewhat insightful

Comment (proteusvacuum) [inline corehq/apps/users/bulkupload.py]: Interesting. I'm still confused as to why we fail when there are capital letters in email addresses though (not why you are adding this, just why we are failing later down the line)

Comment (benrudolph) [inline corehq/apps/users/bulkupload.py]: ah, i misunderstood your question: https://github.com/dimagi/commcare-hq/blob/br/remove-ucr-tours/corehq/apps/users/models.py#L692

Comment (benrudolph) [inline corehq/apps/users/bulkupload.py]: i'm scared to change that as it may have implications on uniqueness assumptions of our emails

Comment (proteusvacuum) [inline corehq/apps/users/bulkupload.py]: Lol. That seems... silly. 
I wonder if forcing it to lowercase is a better option then? Our code did that until August last year:
https://github.com/dimagi/commcare-hq/commit/385ef4316c0093d10fe08cd9cbe39c17f79af49b 
And we do that for forms:
https://github.com/dimagi/commcare-hq/pull/7911/files

Comment (benrudolph) [inline corehq/apps/users/bulkupload.py]: 👍 i'm sold

Comment (benrudolph) [inline corehq/apps/users/bulkupload.py]: https://github.com/dimagi/commcare-hq/pull/12371/commits/676b8c80481e35597c659bc12b00e3d821399bd1

**Comment by proteusvacuum** [comment]
Thanks!
:+1: on :white_check_mark:

**Comment by benrudolph** [comment]
✔️

**Pull request merged**